catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -145,71 +138,88 @@ public String getId() {
+      final boolean hasLocation = hasValidLocation(metacard.getLocation());
+      // The chipping transformer requires the NITF resource, the overview derived image
+      // resource, and the original derived image resource.
+      final boolean hasNitfResource = metacard.getResourceURI() != null;
+      boolean hasOriginalDerivedImageResource = getOriginalDerivedResourceUri(metacard).isPresent();
+      // assume that if there is an original, there is also an overview
+      boolean hasOverviewDerivedImageResource = hasOriginalDerivedImageResource;
+
+      return isImageNitf
+          && hasLocation
+          && hasNitfResource
+          && hasOverviewDerivedImageResource
+          && hasOriginalDerivedImageResource;
+    } else {
+      return false;
     }
-    return false;
   }
 
   private boolean hasValidLocation(String location) {
-    boolean hasValidLocation = false;
-
     if (StringUtils.isNotBlank(location)) {
       try {
         // parse the WKT location to determine if it has valid format
-        WKTReader wktReader = new WKTReader(geometryFactory);
+        final WKTReader wktReader = new WKTReader(geometryFactory);
         wktReader.read(location);
-        hasValidLocation = true;
+        return true;
       } catch (ParseException e) {
-        LOGGER.debug("Location [{}] is invalid, cannot chip this image", location);
+        LOGGER.debug("Location [{}] is invalid. Cannot chip this image", location);
+      }
+    }
+
+    return false;
+  }
+
+  private static Optional<URI> getOriginalDerivedResourceUri(final Metacard metacard) {
+    final Attribute derivedResourceUriAttribute =
+        metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
+    if (derivedResourceUriAttribute != null) {
+      for (Serializable value : derivedResourceUriAttribute.getValues()) {
+        if (value instanceof String) {
+          final String derivedResourceUriString = (String) value;
+          try {
+            final URI derivedResourceUri = new URI(derivedResourceUriString);
+
+            if (shouldBeChippedLocally(derivedResourceUri)) {
+              final String fragment =
+                  derivedResourceUri.getFragment(); // {@code null} if the fragment is undefined
+
+              if (StringUtils.equals(ORIGINAL_QUALIFIER, fragment)) {
+                return Optional.of(derivedResourceUri);
+              }
+            } else {
+              for (NameValuePair parameter :
+                  URLEncodedUtils.parse(derivedResourceUri, StandardCharsets.UTF_8.name())) {
+                if (QUALIFIER_KEY.equals(parameter.getName())) {
+                  final String parameterValue = parameter.getValue();
+
+                  if (StringUtils.equals(ORIGINAL_QUALIFIER, parameterValue)) {
+                    return Optional.of(derivedResourceUri);
+                  }
+                }
+              }
+            }
+          } catch (URISyntaxException e) {
+            // ignore
```

Comment:
✏️  Is this URISyntax exception only thrown from line 187?  If so is there any reason we don't want to just try that line and then `return Optional.empty()` in the catch instead of using the jump to the catch block to ignore the rest of the code?

Reply:
I was trying to be clear that even though there are 2 paths for local and remote chipping, the default value is `Optional.empty()` if the attempt to construct the URL fails.

Reply:
It just seems more concise to state that we return optional.empty by returning optional.empty right where the check happens -- instead of catching it , jumping here, falling through and then returning it at the bottom of the method. It still relays that regardless of the path the default failure is an empty optional since it happens before the check for local/remote.

Reply:
I played around with it a little bit and decided to stick with what I got. I added some more comments though:+1: